### PR TITLE
Modern Relay

### PR DESCRIPTION
**Pull request by Neitsch**
Using Modern Relay with the drop in replacement. I'll keep this commit open to gradually migrate to Modern Relay.
Also
* Added prop-types package
* Changed name to build without problems
* Changed react-router-relay dependency to support the new relay version

### CODE REVIEW

**Comment by lvarayut** [comment]
Thanks @Neitsch, that's looking good to me.

**Comment by ncrmro** [comment]
Just saw this, really excited as well. I'm wondering what the routing story is with regards to react-router 4 as well?

**Comment by Neitsch** [comment]
The issue with `react-router-relay` is that `react-relay` is not backward compatible as is. I submitted a PR, but I don't think there will be a quick solution (relay-tools/react-router-relay#235). In the end Modern Relay is supposed to be somewhat different in routing, so we'll see, whether a new solution emerges or the router is maybe not even necessary any more.

**Comment by Neitsch** [comment]
So, this should be 90% done. Right now we're still in compat mode, since the full switch did not work yet. Nevertheless, it's looking surprisingly good overall. The biggest issue before a full migration is getting the router out of the way.
The one thing that is not very elegant is how I am passing Relay around for the mutation. Somehow the prop validation failed at first, so I relaxed the constraints.

**Comment by Neitsch** [comment]
Anybody who is looking for a reference implementation, look at this https://github.com/relayjs/relay-examples/tree/master/todo-modern

**Comment by Neitsch** [comment]
@ncrmro Might as well just rewrite react-router-relay, since the package is still stuck on react-router 2.8...

File: package.json [hunk 1]
```diff
@@ -83,14 +86,15 @@
...
     "react": "^15.3.2",
     "react-dom": "^15.3.2",
     "react-dropdown": "^1.1.0",
     "react-hot-loader": "3.0.0-beta.6",
     "react-mdl": "^1.7.2",
-    "react-relay": "^0.9.3",
+    "react-relay": "dev",
     "react-router": "^2.8.1",
-    "react-router-relay": "^0.13.5",
+    "react-router-relay": "https://github.com/Neitsch/react-router-relay#fullstack",
```
Comment (Neitsch) [inline package.json:97]: I'm kind of hesitant to merge this in, but since it is canary, I guess it is okay...

**Pull request merged**